pkg/destroy/gcp/instance.go

```diff
@@ -24,7 +24,16 @@ func (o *ClusterUninstaller) getInstanceNameAndZone(instanceURL string) (string,
 }
 
 func (o *ClusterUninstaller) listInstances() ([]cloudResource, error) {
-	return o.listInstancesWithFilter("items/*/instances(name,zone,status),nextPageToken", o.clusterIDFilter(), nil)
+	byName, err := o.listInstancesWithFilter("items/*/instances(name,zone,status),nextPageToken", o.clusterIDFilter(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	byLabel, err := o.listInstancesWithFilter("items/*/instances(name,zone,status),nextPageToken", o.clusterLabelFilter(), nil)
+	if err != nil {
+		return nil, err
+	}
+	return append(byName, byLabel...), nil
```

Comment:
You want to dedup this, right?  Because it seems possible that you'd have an instance that would match both the name prefix and the label.

Reply:
This is deduped when added to the list of `pendingItems` to be deleted.
https://github.com/openshift/installer/blob/115d3d26d1c135cb75f53ba826f5474f6c56471b/pkg/destroy/gcp/gcp.go#L291

Reply:
They are dedup when added to the `pendingItems` list prior to deletion.
https://github.com/openshift/installer/blob/115d3d26d1c135cb75f53ba826f5474f6c56471b/pkg/destroy/gcp/gcp.go#L291-L299
https://github.com/openshift/installer/blob/115d3d26d1c135cb75f53ba826f5474f6c56471b/pkg/destroy/gcp/cloudresource.go#L15-L20